c7n/resources/opensearch.py

```diff
@@ -0,0 +1,118 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+from c7n.manager import resources
+from c7n.query import QueryResourceManager, TypeInfo
+from c7n.tags import RemoveTag, Tag, TagActionFilter, TagDelayedAction, universal_augment
+from c7n.utils import local_session, type_schema
+from c7n.actions import BaseAction
+
+
+@resources.register('opensearch-serverless')
+class OpensearchServerless(QueryResourceManager):
+    class resource_type(TypeInfo):
+        service = 'opensearchserverless'
+        arn_type = 'arn'
+        enum_spec = ('list_collections', 'collectionSummaries[]', None)
+        name = "name"
+        id = "id"
+        cfn_type = 'AWS::OpenSearchServerless::Collection'
+        arn = "arn"
+        permission_prefix = 'aoss'
+
+    def augment(self, resources):
+        return universal_augment(self, super().augment(resources))
+
+
+@OpensearchServerless.action_registry.register('tag')
+class TagOpensearchServerlessResource(Tag):
+    """Create tags on an OpenSearch Serverless resource
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+            - name: tag-opensearch-serverless
+              resource: opensearch-serverless
+              actions:
+                - type: tag
+                  key: test-key
+                  value: test-value
+    """
+    permissions = ('aoss:TagResource',)
```

Comment:
if the resource supports resource group tagging, why implement the actions against the  service tagging api versus using the resource group tagging actions? Ala adding `universal_taggable = object()` to the resource_type should pick these up and allows bulk tagging operations.

Reply:
It doesn't seem to be supported in resourcegrouptagging api. I tried to tag a test resource using the below api and it didn't throw an error but the tags didn't go through
```
response = client.tag_resources(ResourceARNList=['arn:aws:aoss:us-east-1:644160558196:collection/z0a6l95oofv72h8qzinj'], Tags={'Owner': 'Pratyush'})
```

Reply:
I was running into an error when trying that. This error was outputting:
`Exception: Resource Tag Errors {'arn:aws:aoss:......': InvalidParameterException}`